zjit/src/codegen.rs

```diff
@@ -439,6 +439,24 @@ fn gen_get_ep(asm: &mut Assembler, level: u32) -> Opnd {
     ep_opnd
 }
 
+fn gen_objtostring(jit: &mut JITState, asm: &mut Assembler, val: Opnd, cd: *const rb_call_data, state: &FrameState) -> Option<Opnd> {
+    gen_prepare_non_leaf_call(jit, asm, state)?;
```

Comment:
I couldn't figure out which part of `rb_vm_objtostring` calls a method. Could you point out which part of it is not leaf?

Reply:
When `recv.singleton_class?` rb_mod_to_s() calls `rb_any_to_s(recv.attached_object)`

Reply:
Yes, in general `PRIsVALUE` can call `to_s` by going through `PRI_EXTRA_MARK` and ruby__sfvextra(). Only for non-T_STRINGs does ruby__sfvextra() make a call, though, and it looks like `cname` is always a string here. Though there is also an encoding check in there so the print might rise.